### ISSUE TITLE
could you release the train/val/test lists

### ISSUE THREAD

**Issue by ainnn** [chiutaiyin/VizWiz-QualityIssues#2]
1. could you release the train/val/test lists used in sec.4.2 (Table 1) for result reproduction and fair comparison?
2. how to get recall and f1 in table 1. is it that recall=TP/(TP+FN), f1=2*recall*precision/(recall+precision) with precision=TP/(TP+FP) which is different from the reported average precision?
thank you.

**Comment by ainnn** [comment]
the sigmoid output is the probability of recognizable and when i say POSITIVE i mean unrecognizable. so when the sigmoid value is less than 0.5, we predict that the label is unrecognizable(POSITIVE). am i right?

**Comment by ainnn** [comment]
more questions. 
1. for sec.4, why we need so many val data (37.5%)? what are they for?
2. are results in table 1. reported from test set (10%)?
3. if you cannot provide the data split files, could you provide the recognizable/unrecognizable sample amount(ratio) used in table 1.
thank you.

**Comment by chiutaiyin** [comment]
> 1. could you release the train/val/test lists used in sec.4.2 (Table 1) for result reproduction and fair comparison?
> 2. how to get recall and f1 in table 1. is it that recall=TP/(TP+FN), f1=2_recall_precision/(recall+precision) with precision=TP/(TP+FP) which is different from the reported average precision?
>    thank you.

Hi,
For your first question, let me explain why we tend not to release the train/val/test split used in Table 1 after some consideration (but we have another way to maintain reproductivity): since we reorganized the VizWiz dataset, the current version is a bit different from the one we used in the paper. The test set used in Table 1 was only 10% of the old training set and was composed of roughly 1500 images, and we think developing an algorithm under such a small (and potentially biasedly selected) test set is not good. Instead, we are now building an online evaluation system on EvalAI, where you can submit your whole test results as many times as you want. It will tell you the performance on test-dev set of your algorithm, which is supposed to be developed under the whole VizWiz dataset. Then you can compare your performance with the one reported on the project website (https://github.com/chiutaiyin/VizWiz-QualityIssues).

For the second question, yes, your interpretation is right. Just note that f1 is computed using precision but not average precision. Or for simplicity, you can just stick with average precision (or f1) as your main metric.

**Comment by chiutaiyin** [comment]
> the sigmoid output is the probability of recognizable and when i say POSITIVE i mean unrecognizable. so when the sigmoid value is less than 0.5, we predict that the label is unrecognizable(POSITIVE). am i right?

Yes, you're right!

**Comment by chiutaiyin** [comment]
> more questions.
> 
> 1. for sec.4, why we need so many val data (37.5%)? what are they for?

Yes, you are right. It probably doesn't need that much data for validation.

> 2. are results in table 1. reported from test set (10%)?

Yes
> 3. if you cannot provide the data split files, could you provide the recognizable/unrecognizable sample amount(ratio) used in table 1.

Since each split is randomly sampled from the old training dataset, where about 15% of images are unrecognizable, the ratio of unrecognizable images in each split follows that statistic.

Thank you.

**Comment by ainnn** [comment]
thanks for your detailed response.
in my experience, class balancing affects the test results. my mobilenet model got ap=0.91, f1=0.83 with unrecognizable sample ratio 975/1950 (unrec/(unrec+rec)), but only got ap=0.52, f1=0.64 with the ratio 233/1500. thus achieving an ap of 80.0% on a biased test set is harder than on a unbiased one.
looking forward to  the online evaluation system. thank you.
